### DISCUSSION TITLE
Pushing an image to private regsitry using golang package

### DISCUSSION THREAD

**Discussion by imaskm** [containerd/nerdctl#1159] [Q&A] [answered]
Hi, how can we push an image to a private registry  using the package   "github.com/containerd/nerdctl/pkg/imgutil/push",

I didn't find any option to pass for authentication.

**Comment by junnplus** [comment, marked as answer]
You can construct a resolver [with authentication](https://github.com/containerd/nerdctl/blob/master/pkg/imgutil/dockerconfigresolver/dockerconfigresolver.go#L85-L89), using `$DOCKER_CONFIG/config.json` as [authentication](https://github.com/containerd/nerdctl/blob/master/pkg/imgutil/dockerconfigresolver/dockerconfigresolver.go#L166) by default

**Reply by imaskm** [reply]
@junnplus  thanks for quick response.
